pkg/flow/streaming/flow.go

```diff
@@ -0,0 +1,175 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package streaming
+
+import (
+	"context"
+	"io"
+	"reflect"
+
+	"github.com/pkg/errors"
+
+	"github.com/apache/skywalking-banyandb/pkg/flow"
+	streamingApi "github.com/apache/skywalking-banyandb/pkg/flow/streaming/api"
+	"github.com/apache/skywalking-banyandb/pkg/flow/streaming/sources"
+)
+
+var _ flow.Flow = (*streamingFlow)(nil)
+
+type streamingFlow struct {
+	// sourceParam and sinkParam are generic source and sink
+	sourceParam interface{}
+	sinkParam   interface{}
+
+	ctx    context.Context
+	source streamingApi.Source
+	sink   streamingApi.Sink
+	ops    []streamingApi.Operator
+	drain  chan error
```

Comment:
Test cases have been added to verify this error chan while a single goroutine is started to consume this channel to avoid too many waiting goroutines.

`TearDown` is also used to ensure resource recycling during the shutdown.

Reply:
Where to close this chan?

Reply:
`drainErr` function has been switched to sync mode, which means error channel must be properly handled after `Open`.

So we can safely close `errorChan` after all components are closed.